presto-orc/src/main/java/com/facebook/presto/orc/DecodeTimestampOptions.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc;
+
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+
+import java.util.concurrent.TimeUnit;
+
+import static java.util.Objects.requireNonNull;
+
+public class DecodeTimestampOptions
+{
+    private static final int MILLIS_PER_SECOND = 1000;
+
+    private final long timestampUnitsPerSecond;
+    private final long nanosecondsPerTimeUnit;
+    private final long baseTimestampInSeconds;
+
+    public DecodeTimestampOptions(
+            DateTimeZone hiveStorageTimeZone, TimeUnit timestampTimeUnit)
+    {
+        this.timestampUnitsPerSecond = requireNonNull(timestampTimeUnit).convert(1, TimeUnit.SECONDS);
```

Comment:
`timeStampTimeUnit.compareTo(TimeUnit.SECONDS) <=0`

Reply:
I wanted to materialize nanosecondsPerTimeUnit/timestampUnitsPerSecond in ctor to avoid multiple virtual function calls on critical path (ApacheHiveTimestampDecoder#decodeTimestamp). Do you think it's ok?

Reply:
thanks, you are right.